python/sedUtils.py

```diff
@@ -0,0 +1,273 @@
+
+
+def readPhotSeds(sedDir=None):
+    """Read all the seds provided by this package, storing them in a
+    dictionary and saving lists of the SEDs of each type (so that
+    they can be separated later if desired). """
+    # The environment variable PHOT_REF_SEDS_DIR is set if the file
+    #  'phot_ref.csh' is sourced.
+    if sedDir == None:
+        sedDir = os.path.join(os.getenv('SYSENG_THROUGHPUTS_DIR'), 'seds')
+    # Set up the lists for the photometry reference SEDs.
+    #  (I'm doing this by hand to keep these lists well-known over time.)
+    sedlists = {}
+    sedlists['quasar'] = ['quasar.dat',]
+    sedlists['stars'] = ['km10_7250.fits_g45', 'km10_6500.fits_g45',
+                         'km10_6000.fits_g45', 'km10_5250.fits_g45',
+                         'km10_4500.fits_g45', 'm3.0Full.dat']
+    sedlists['white_dwarf'] = ['wd_H_100000_80.dat', 'wd_H_15000_80.dat', 'wd_H_50000_80.dat',
+                               'wd_H_5500_80.dat', 'wd_He_10000_80.dat', 'wd_He_15000_80.dat', 'wd_He_5500_80.dat']
+    sedlists['sn'] = ['sn1a_15.0.dat', 'sn1a_20.0.dat', 'sn1a_10.0.dat']
+    sedlists['galaxies'] = ['Sa_template_norm.sed.dat', 'Sdm_template_norm.sed0.dat',
+                            'Ell2_template_norm.sed.dat']
+    sedlists['photoZ_outliers'] = ['xspec_172.sed.dat', 'xspec_173.sed.dat',
+                                   'xspec_175.sed.dat', 'xspec_176.sed.dat',
+                                   'xspec_90.sed.dat', 'xspec_91.sed.dat']
+    # Let's go read the files.
+    sedDict = {}
+    # Loop through quasar, stars, galaxies, sn in the sedlist dictionary.
+    for objtype in sedlists.keys():
+        # And for each type of object, loop through and read each SED.
+        for s in sedlists[objtype]:
+            sedDict[s] = Sed()
+            sedDict[s].readSED_flambda(os.path.join(sedDir, objtype, s))
+    # Return the sed dictionary and the dictionary containing the lists of each
+    #  type of object.
+    return sedDict, sedlists
+
+def readAnySeds(inputfileList, sedDir=None):
+    """Read the seds in a list and store them in a format appropriate for use with the
+    rest of these routines. sedDir (if set) can be the root directory for the files in the list."""
+    # Set up the sedlists dictionary to hold the sed names. We'll store them all keyed under 'any'.
+    sedlists = {}
+    sedlists['any'] = deepcopy(inputfileList)
+    # If the root directory is set, add it to the input file names.
+    if sedDir != None:
+        ifiles = []
+        for i in inputfileList:
+            ifiles.append(os.path.join(sedDir, i))
+        inputfileList = ifiles
+    # Read the files.
+    sedDict = {}
+    for filename, s in zip(inputfileList, sedlists['any']):
+        sedDict[s] = Sed()
+        sedDict[s].readSED_flambda(filename)
+    # Return the sed dictionary and the dictionary containing the lists of each type of object.
+    return sedDict, sedlists
+
+
+def makeRedshiftedSeds(sedDict, sedlists, redshifts):
```

Comment:
Okay: sedlists is a dict of lists of sed names.  Is it possible to just work with discrete sedDicts (one for quasars, one for galaxies, one for stars).  Or even just make sedDict a dict of dicts, so that

sedDict ['quasar']['quasar_name'] 

was a quasar SED?  That way, you wouldn't have to keep carrying around both sedDict and sedlists